R/mda_processes.R

```diff
@@ -79,6 +79,22 @@ create_mda_listeners <- function(
       variables$drug_time$queue_update(timestep, to_move)
     }
 
+    # Update liver stage drug effects
+    if(length(parameters$drug_hypnozoite_efficacy)>0){
+
+      successful_hyp_treatments <- bernoulli(
+        length(target),
+        parameters$drug_hypnozoite_efficacy[[drug]]
+      )
+
+      to_clear <- individual::Bitset$new(parameters$human_population)
+      to_clear$insert(target[successful_hyp_treatments])
```

Comment:
We have a faster implementation of this in utils.R

```suggestion
      to_clear <- sample_bitset(
        target,
        parameters$drug_hypnozoite_efficacy[[drug]]
      )
```

Reply:
Oh cool - I had basically just copied the blood stage prophylaxis code in these sections. I'm guessing that these can also be changed to use the sample_bitset function. e.g., in mda_processes:
![image](https://github.com/mrc-ide/malariasimulation/assets/73181123/aca93750-d142-483d-90a8-2241d93a5817)

I'll make the same change to these parts, assuming there isn't a reason this code needs to be different.

Reply:
I've made the change, but have had to leave in the creation of a new bitset (`target_bit`) because the current target is only a vector. I think it's still more efficient (at least in terms of space).